From d540a423bf5dd6c2d10680031b644eb607f8867d Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Sun, 26 May 2019 20:04:51 +0200
Subject: [PATCH 183/345] MIPS: OCTEON: Allow CONFIG_CAVIUM_CN63XXP1 to be
 disabled.

It is advantageous to use CONFIG_CAVIUM_OCTEON2 with OCTEON III CPU
cores as they have the same instruction set.  However, OCTEON III
doesn't require the CN63XXP1 workaround, so make it optional.

Signed-off-by: David Daney <david.daney@cavium.com>
---
 arch/mips/cavium-octeon/Kconfig | 15 +++++++--------
 1 file changed, 7 insertions(+), 8 deletions(-)

diff --git a/arch/mips/cavium-octeon/Kconfig b/arch/mips/cavium-octeon/Kconfig
index 9cb2a427e6ce..cd044186e151 100644
--- a/arch/mips/cavium-octeon/Kconfig
+++ b/arch/mips/cavium-octeon/Kconfig
@@ -2,14 +2,13 @@
 if CPU_CAVIUM_OCTEON
 
 config CAVIUM_OCTEON2
-       bool "Build the kernel to be used only on OCTEON II processor cores"
-       select CAVIUM_CN63XXP1
-       default "n"
-       help
-	 This option enables the generation of Octeon2 specific
-	 instructions by the compiler, resulting in a kernel that is
-	 more efficient, but that will not run on Octeon and
-	 OcteonPlus processor cores.
+	bool "Build the kernel to be used only on OCTEON II or OCTEON III processor cores"
+	default "n"
+	help
+	  This option enables the generation of Octeon2 specific
+	  instructions by the compiler, resulting in a kernel that is
+	  more efficient, but that will not run on Octeon and
+	  OcteonPlus processor cores.
 
 config CAVIUM_CN63XXP1
 	bool "Enable CN6XXX DCache prefetch errata workaround"
-- 
2.25.1

